docs/pre-release/argo/canary-rollouts-with-argo.md

```diff
@@ -0,0 +1,262 @@
+```
+
+Now, install the Argo Rollouts controller in your cluster. Note that Argo must be installed in the `argo-rollouts` namespace.
+
+```
+kubectl create namespace argo-rollouts
+kubectl apply -n argo-rollouts -f https://raw.githubusercontent.com/datawire/argo-rollouts/ambassador/deploy/manifests/install.yaml
+```
+
+Finally, install the `kubectl` plugin, which will let you manage and visualize rollouts from the command line.
+
+Mac OS X:
+
+```
+brew install argoproj/tap/kubectl-argo-rollouts
+```
+
+Linux:
+
+```
+sudo curl -fL https://github.com/argoproj/argo-rollouts/releases/latest/download/kubectl-argo-rollouts-linux-amd64 -o /usr/local/bin/kubectl-argo-rollouts
+sudo chmod a+x /usr/local/bin/kubectl-argo-rollouts
+```
+
+Verify correct installation:
+
+```
+kubectl argo rollouts version
+```
+
+## 3. Create the Kubernetes Services
+
+We'll create two Kubernetes services, named `echo-stable` and `echo-canary`:
+
+```
+apiVersion: v1
+kind: Service
+metadata:
+  labels:
+    app: echo
+  name: echo-stable
+spec:
+  type: ClusterIP
+  ports:
+  - name: http
+    port: 80
+    protocol: TCP
+    targetPort: 8080
+  selector:
+    app: echo
+---
+apiVersion: v1
+kind: Service
+metadata:
+  labels:
+    app: echo
+  name: echo-canary
+spec:
+  type: ClusterIP
+  ports:
+  - name: http
+    port: 80
+    protocol: TCP
+    targetPort: 8080
+  selector:
+    app: echo 
+```
+
+We'll also create an Edge Stack route to the services:
+
+```
+apiVersion: getambassador.io/v2
+kind:  Mapping
+metadata:
+  name:  rollouts-demo
+spec:
+  prefix: /demo
+  rewrite: /demo
+  service: echo-stable:80
+  resolver: endpoint
+```
+
+## 4. Deploy the Echo Service
+
+Argo Rollouts is orchestrated via the `Rollouts` resource. A `Rollout` resource is a superset of the standard Kubernetes `deployment` resource, and adds the ability to manage the process by which your service is deployed.
+
+Create a `Rollout` resource and save it to a file called `rollout.yaml`:
+
+```
+apiVersion: argoproj.io/v1alpha1
+kind: Rollout
+metadata:
+  name: echo-rollout
+spec:
+  selector:
+    matchLabels:
+      app: echo
+  template:
+    metadata:
+      labels:
+        app: echo
+    spec:
+      containers:
+        - image: hashicorp/http-echo
+          args:
+            - "-text=VERSION 1"
+            - -listen=:8080
+          imagePullPolicy: Always
+          name: echo-v1
+          ports:
+            - containerPort: 8080
+  strategy:
+    canary:
+      stableService: echo-stable
+      canaryService: echo-canary
+      trafficRouting:
+        ambassador:
+          mapping: echo
+      steps:
+      - setWeight: 30
+      - pause: {duration: 30s}
+      - setWeight: 60
+      - pause: {duration: 30s}
+      - setWeight: 100
+      - pause: {duration: 10}
+```
+
+Apply the rollout to your cluster `kubectl apply -f rollout.yaml`. Note that no canary rollout will occur, as this is the first version of the service being deployed.
```

Comment:
point out that the strategy.canary section contains a trafficRouting property using ambassador and the echo mapping we configured above!?